design-proposals/cluster-orch-integrating-k8s-into-emt.md

```diff
@@ -0,0 +1,218 @@
+
+With this design change, we anticipate reduced and consistent cluster creation times, enhanced reliability by eliminating third-party dependencies, improved security through faster application of CVEs, and an improved user experience by providing a zero-touch option from device onboarding to a cluster-ready state.
+
+## Proposal
+
+### Building Kubernetes
+
+#### What to build and package
+
+Cluster Orchestration supports two flavors of Kubernetes distributions: RKE2 and K3s [todo: add link to k3s design proposal]. In the initial implementation, we will focus on K3s, as it is set to be the default Kubernetes distribution in Cluster Orchestration. K3s is simpler to build compared to RKE2 because it integrates Kubernetes into a single binary, making it a preferred choice for quick implementation.
+
+The build process will focus on creating the first-level of binary and assets, excluding their dependencies. For instance, the K3s build process involves downloading and building containerd, etcd, and other dependencies from the upstream repositories. We will leverage their existing build process without building these nested dependency separately. For a comprehensive list of dependencies that we are not directly building, refer to the [K3s build script](https://github.com/k3s-io/k3s/blob/master/scripts/version.sh). This decision may evolve based on future needs.
+
+Assets to be built and packaged for K3s installation include the following. The list of container image is sourced from https://github.com/k3s-io/k3s/blob/master/scripts/airgap/image-list.txt. In the initial implementation, we will focus on building the K3s binary and eventually extend to building container images for addons.
+
+- k3s (binary)
+- k3s-selinux (rpm)
+- klipper-helm (container image)
+- klipper-lb (container image)
+- local-path-provisioner (container image)
+- coredns (container image)
+- busybox (container image)
+- traefik (container image)
+- metrics-server (container image)
+- pause (container image)
+- install.sh (script)
+
+For simplicity, the K3s version will be limited to the latest stable release, and users will not be able to select a specific version. A new EMT image will be released under two additional conditions: when a new K3s version becomes available or when a critical CVE or security patch is required, supplementing the existing release cadence. This decision may evolve based on future user requirements.
```

Comment:
Doesn't this conflict with cluster templates which specify a k8s version to be installed.

Reply:
Addressed in later in Manual Cluster Creation section